components/datadog/dockeragentparams/params.go

```diff
@@ -0,0 +1,188 @@
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/test-infra-definitions/common"
+	"github.com/DataDog/test-infra-definitions/components/command"
+	"github.com/DataDog/test-infra-definitions/components/datadog/fakeintake"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+// Params defines the parameters for the Docker Agent installation.
+// The Params configuration uses the [Functional options pattern].
+//
+// The available options are:
+//   - [WithImageTag]
+//   - [WithRepository]
+//   - [WithPulumiDependsOn]
+//   - [WithEnvironmentVariables]
+//   - [WithAgentServiceEnvVariable]
+//   - [WithHostName]
+//	 - [WithFakeintake]
+//	 - [WithLogs]
+//   - [WithExtraComposeManifest]
+//
+// [Functional options pattern]: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+
+type Params struct {
+	// ImageTag is the docker agent image tag to use.
+	ImageTag string
+	// Repository is the docker repository to use.
+	Repository string
+	// AgentServiceEnvironment is a map of environment variables to set in the docker compose agent service's environment.
+	AgentServiceEnvironment pulumi.Map
+	// ExtraComposeManifests is a list of extra docker compose manifests to add beside the agent service.
+	ExtraComposeManifests []command.DockerComposeInlineManifest
+	// EnvironmentVariables is a map of environment variables to set with the docker-compose context
+	EnvironmentVariables pulumi.StringMap
+	// PulumiDependsOn is a list of resources to depend on.
+	PulumiDependsOn []pulumi.ResourceOption
+}
+
+type Option = func(*Params) error
+
+func NewParams(options ...Option) (*Params, error) {
+	version := &Params{
+		AgentServiceEnvironment: pulumi.Map{},
+		EnvironmentVariables:    pulumi.StringMap{},
+	}
+	return common.ApplyOption(version, options)
+}
+
+func WithImageTag(agentImageTag string) func(*Params) error {
+	return func(p *Params) error {
+		p.ImageTag = agentImageTag
+		return nil
+	}
+}
+
+func WithRepository(repository string) func(*Params) error {
+	return func(p *Params) error {
+		p.Repository = repository
+		return nil
+	}
+}
+
+func WithPulumiDependsOn(resources ...pulumi.ResourceOption) func(*Params) error {
+	return func(p *Params) error {
+		p.PulumiDependsOn = resources
+		return nil
+	}
+}
+
+func WithEnvironmentVariables(environmentVariables pulumi.StringMap) func(*Params) error {
+	return func(p *Params) error {
+		p.EnvironmentVariables = environmentVariables
+		return nil
+	}
+}
+
+// WithAgentServiceEnvVariable set an environment variable in the docker compose agent service's environment.
+func WithAgentServiceEnvVariable(key string, value pulumi.Input) func(*Params) error {
+	return func(p *Params) error {
+		p.AgentServiceEnvironment[key] = value
+		return nil
+	}
+}
+
+// WithIntake configures the agent to use the given hostname as intake.
+//
+// To use a fakeintake, see WithFakeintake.
+//
+// This option is overwritten by `WithFakeintake`.
+func WithIntake(hostname string, skipSSL bool) func(*Params) error {
```

Comment:
`skipSSL` is not very clear, it would be more something like `tlsVerify`. But honestly you can forget about the parameter entirely, we'll ~never have truely valid certificate.

Reply:
In containers test currently we use double shipping to real DD and fakeintake by default. I imagine it's going to be the case for docker based tests, and it could be the case for VM based tests, as it makes it easier to investigate or test cross intake. Will change to `tlsVerify` as suggested !

Reply:
Ok so it is for DD target. I think we should rework a bit. Unfortunately the endpoint configuration is the Agent is really messy, but currently you hardcode `http` in some cases while you check if target is `https` for logs?

I don't think we want to support mixed http/https on the same endpoint.

Also you're checking `https` for `DD_LOGS_CONFIG_FORCE_USE_HTTP` but `DD_LOGS_CONFIG_FORCE_USE_HTTP` is to switch from TCP to HTTP/HTTPS. AFAIK logs with HTTP for main endpoint but not for additional endpoints? @L3n41c to confirm as you looked at it recently.

Reply:
Ah actually we have a `WithFaketintake` and `WithExtraIntake`, we can keep the ssl validation hidden indeed.

Reply:
Changed !